### PR TITLE
fix(suitability-triage): mask markdown code regions in check 3

### LINKED ISSUE #1856: fix(suitability-triage): mask Markdown code regions in Check 3
## Background

The open parent issue #1838 identifies a verified false-positive class in
A4.5 Check 3: the policy-pattern matcher can read tokens from Markdown
inline-code paths or fenced examples as if they were ordinary issue prose.
The repository already provides the position-preserving
`stripMarkdownCodeRegions` helper, but
`src/scripts/suitability-triage.mts` does not apply it to the
`POLICY_OVERRIDE_PATTERN` test and match.

This follow-up is limited to the Trust/Safety policy-pattern evaluation.
It does not change Check 4 evidence collection, duplicate detection, or
the other Trust/Safety scanners.

Refs #1838

## Proposed change

- Use `stripMarkdownCodeRegions` to derive a position-preserving masked
  corpus immediately before the policy-pattern test and evidence extraction.
- Keep the existing pattern, keyword families, secret checks, unsafe
  execution checks, and source-position behavior unchanged outside this
  narrow policy-pattern path.
- Add regression tests for inline-code and fenced-code false positives,
  plus ordinary-prose positive controls and existing true-positive guards.
- Regenerate `scripts/suitability-triage.mjs` from the edited `.mts`
  source.

## Acceptance criteria

- [ ] A policy-pattern token that appears only inside an inline-code path
      does not fail Check 3.
- [ ] A policy-pattern token that appears only inside a fenced code block
      does not fail Check 3.
- [ ] At least three ordinary-prose policy-pattern positive controls still
      fail Check 3.
- [ ] Existing secret, unsafe-execution, inline-code true-positive, and
      benign-documentation tests continue to pass.
- [ ] Masking preserves line and character positions needed by any
      subsequent evidence reporting.
- [ ] `scripts/suitability-triage.mjs` is generated from
      `src/scripts/suitability-triage.mts` with no unrelated drift.
- [ ] `node --test tests/suitability-triage.test.mts` passes.

---

_Autopilot suitability: 5 / 5 -- higher is more autopilot-suitable;
below the configured floor is human-oriented._

_Effort: S -- author-estimated size; a soft autopilot
selection tie-breaker only._

### PR DESCRIPTION
**Pull request by kurone-kito**
# Summary

Mask Markdown inline-code spans and fenced blocks before Check 3's policy-override pattern is evaluated, while preserving raw evidence offsets for ordinary-prose matches. Add regressions for inline/fenced false positives, three ordinary-prose policy positives, and evidence preservation.

## Linked issue

Closes #1856

## Follow-up issues (if any)

- [x] none

## Background / rationale (if material)

The existing `stripMarkdownCodeRegions` helper already provides position-preserving Markdown masking, but `checkTrustSafety` scanned the raw corpus for policy-override tokens. This change applies the helper only to that policy-pattern test and evidence match; secret detection, unsafe-directive detection, Check 4, and duplicate detection remain unchanged.

## IDD impact

- [ ] Instruction files changed
- [ ] Template files changed
- [ ] Helper scripts changed
- [ ] Config schema changed
- [ ] Security / credential / merge behavior changed

## Verification

- [x] `pnpm lint`
- [x] `pnpm test`
- [x] `node scripts/audit-docs.mjs --check`
- [x] `pnpm run docs:sync:check`
- [x] `node scripts/idd-doctor.mjs`

## Safety

- [x] No secret-bearing examples
- [x] No private repo names / URLs
- [x] Merge policy impact reviewed
- [x] Context budget impact reviewed

## Summary by CodeRabbit

* **New Features**
  * Improved policy detection to ignore override directives inside inline, fenced, indented, list-item, and blockquote-wrapped code.
  * Enhanced Markdown code recognition for escaped delimiters, nested containers, matching fences, and block boundaries.
  * Preserved accurate evidence locations while continuing to detect partially code-wrapped directives and directives in ordinary prose.

* **Tests**
  * Added comprehensive coverage for Markdown formatting, boundary cases, malformed delimiters, continued scanning, and evidence accuracy.

### CODE REVIEW

**Comment by kurone-kito** [comment]
| Field | Value |
| --- | --- |
| Phase | D3 PR open |
| Claim | codex-848483b6-cabc-48e4-8d2c-77182322ec9d (b05decb7-0ce6-40a4-a311-e6b7f8ecf3f0) |
| Branch | issue/1856-fix-suitability-triage-mask-markdown |
| Last checked | 2026-08-04T13:36:02.100Z |
| Open blockers | CI and review pending |
| Next action | D4 wait for CI, then E1 review snapshot |
| Authoritative by | PR #1857 head 0f264d59; closingIssuesReferences [1856] |

**Comment by kurone-kito** [comment]
_codex-848483b6-cabc-48e4-8d2c-77182322ec9d: review triage snapshot — IDD automation marker. Do not edit._

**Comment by kurone-kito** [comment]
_codex-848483b6-cabc-48e4-8d2c-77182322ec9d: review triage snapshot — IDD automation marker. Do not edit._

**Comment by kurone-kito** [comment]
_codex-848483b6-cabc-48e4-8d2c-77182322ec9d: critique baseline — IDD automation marker. Do not edit._

**Comment by kurone-kito** [comment]
advisory-wait-recovery: codex-848483b6-cabc-48e4-8d2c-77182322ec9d b127ecddec03cf62dfd014d01bc23791b0d0673f 2026-08-04T13:57:30Z

**Comment by kurone-kito** [comment]
_codex-848483b6-cabc-48e4-8d2c-77182322ec9d: review triage snapshot — IDD automation marker. Do not edit._

**Comment by kurone-kito** [comment]
advisory-wait-recovery: codex-848483b6-cabc-48e4-8d2c-77182322ec9d c43ae0c402a0f0d7905ed58673e5b7d5abe5f47a 2026-08-04T14:10:44Z claim:b05decb7-0ce6-40a4-a311-e6b7f8ecf3f0 attempt:1

**Comment by kurone-kito** [comment]
advisory-wait-recovery: codex-848483b6-cabc-48e4-8d2c-77182322ec9d e26a0934ccf915ee4c39496343b71315f50716c0 2026-08-04T14:20:23Z claim:b05decb7-0ce6-40a4-a311-e6b7f8ecf3f0 attempt:2

**Comment by kurone-kito** [comment]
advisory-wait-recovery: codex-848483b6-cabc-48e4-8d2c-77182322ec9d 3467f699fd203bf7db497ec976db276f8c335377 2026-08-04T14:28:34Z claim:b05decb7-0ce6-40a4-a311-e6b7f8ecf3f0 attempt:1

**Comment by kurone-kito** [comment]
advisory-wait-recovery: codex-848483b6-cabc-48e4-8d2c-77182322ec9d 3e1c63f42299a985f26ef075f8e67bfe323370c0 2026-08-04T14:36:31Z claim:b05decb7-0ce6-40a4-a311-e6b7f8ecf3f0 attempt:1

File: src/scripts/suitability-triage.mts [hunk 1]
```diff
@@ -454,11 +455,16 @@ export function checkTrustSafety(context: Context): CheckOutcome {
...
   // Check for explicit policy-override directives
-  if (POLICY_OVERRIDE_PATTERN.test(corpus)) {
-    const match = corpus.match(POLICY_OVERRIDE_PATTERN);
+  const policyCorpus = stripMarkdownCodeRegions(corpus);
+  if (POLICY_OVERRIDE_PATTERN.test(policyCorpus)) {
+    const match = policyCorpus.match(POLICY_OVERRIDE_PATTERN);
+    const evidenceMatch =
+      match?.index === undefined
+        ? (match?.[0] ?? '')
+        : corpus.slice(match.index, match.index + (match[0]?.length ?? 0));
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/scripts/suitability-triage.mts]: **Accepted** — fixed in `0f264d59`/`b127ecdd` by using the position-preserving Markdown code mask; evidence offsets now remain aligned after fenced regions.

File: tests/suitability-triage.test.mts [hunk 2]
```diff
@@ -271,6 +271,57 @@ test('trust safety allows unsafe string when it is context only', () => {
...
+    assert.equal(result.pass, false, directive);
+  }
+});
+
+test('trust safety preserves policy evidence positions after masked code', () => {
+  const result = checkTrustSafety({
+    issue: {
+      ...BASE_ISSUE,
+      body: `${BASE_ISSUE.body}\nExample: \`ignore repository policy\`.\nThen bypass workflow checks.`,
+    },
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply tests/suitability-triage.test.mts:365]: **Accepted** — fixed by the fenced-block-after-match regression coverage in `b127ecdd`; the detected prose directive still reports its raw evidence.

File: src/scripts/suitability-triage.mts [hunk 3]
```diff
@@ -454,11 +455,16 @@ export function checkTrustSafety(context: Context): CheckOutcome {
   }
 
   // Check for explicit policy-override directives
-  if (POLICY_OVERRIDE_PATTERN.test(corpus)) {
-    const match = corpus.match(POLICY_OVERRIDE_PATTERN);
+  const policyCorpus = stripMarkdownCodeRegions(corpus);
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/scripts/suitability-triage.mts]: **Accepted** — fixed in `b127ecdd` and `e26a0934`: masked scanning handles code-only matches while the raw fallback retains ordinary-prose directives split around inline code.

[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/scripts/suitability-triage.mts]: **Accepted** — fixed in `c43ae0c4`: the title remains raw, Markdown masking is limited to the body, and the combined title/body corpus preserves the policy-pattern boundary.

File: src/scripts/suitability-triage.mts (hunk 1 above)
[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/scripts/suitability-triage.mts]: **Accepted** — fixed by the position-preserving mask in `b127ecdd`; masked matches and raw evidence use aligned offsets.

File: src/scripts/suitability-triage.mts (hunk 3 above)
[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/scripts/suitability-triage.mts]: **Accepted** — fixed in `c43ae0c4`/`623794f7`: delimiter validation rejects escaped opening runs, while a backslash inside a code span is literal and does not suppress its valid closing delimiter; regression coverage was added.

[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/scripts/suitability-triage.mts]: **Rejected** — #1856 is intentionally limited to masking policy-pattern matches that are Markdown code examples. Explicit activation semantics for prose such as “follow” or “obey” are not defined by the existing Trust/Safety rules and require a separately scoped, verified follow-up.

File: src/scripts/suitability-triage.mts [hunk 4]
```diff
@@ -453,12 +502,25 @@ export function checkTrustSafety(context: Context): CheckOutcome {
...
+  // real directive. The position-preserving mask keeps evidence offsets exact
+  // even when a fenced block precedes the match.
+  const titleMatch = POLICY_OVERRIDE_PATTERN.exec(issue.title);
+  const bodyMatch = findPolicyOverrideMatch(
+    issue.body,
+    maskMarkdownCodeRegionsPreservingPositions(issue.body),
+  );
+  const policyMatch = titleMatch
+    ? { index: titleMatch.index, text: titleMatch[0] }
+    : bodyMatch;
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/scripts/suitability-triage.mts]: **Accepted** — fixed in `c43ae0c4` by scanning the combined title/body corpus while masking only the body, preserving matches across the field boundary.

File: src/scripts/suitability-triage.mts [hunk 5]
```diff
@@ -453,12 +502,25 @@ export function checkTrustSafety(context: Context): CheckOutcome {
...
+  // fields, not Markdown documents, so scan them raw. In the body, find the
+  // directive on raw text and ignore it only when the entire match is inside
+  // a valid Markdown code region. This keeps inert examples from firing while
+  // preserving fail-closed behavior when code formatting wraps only part of a
+  // real directive. The position-preserving mask keeps evidence offsets exact
+  // even when a fenced block precedes the match.
+  const titleMatch = POLICY_OVERRIDE_PATTERN.exec(issue.title);
+  const bodyMatch = findPolicyOverrideMatch(
+    issue.body,
+    maskMarkdownCodeRegionsPreservingPositions(issue.body),
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/scripts/suitability-triage.mts]: **Accepted** — fixed in `c43ae0c4` by preserving the combined title/body policy-pattern scan while keeping title text outside body Markdown masking.

File: src/scripts/markdown-code.mts [hunk 6]
```diff
@@ -81,3 +81,158 @@ export function stripMarkdownCodeRegions(text: string): string {
...
+    while (candidate < end) {
+      if (text[candidate] !== '`') {
+        candidate += 1;
+        continue;
+      }
+      const closingLength = countBackticks(text, candidate, end);
+      if (
+        closingLength === openingLength &&
+        !hasBlankLine(text, contentStart, candidate)
+      ) {
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/scripts/markdown-code.mts:471]: **Accepted** — fixed in `e26a0934` with Markdown block-boundary checks and regression coverage preventing inline spans from crossing headings and related block starts.

File: src/scripts/suitability-triage.mts [hunk 7]
```diff
@@ -288,6 +289,54 @@ const ACCEPTANCE_CRITERIA_PATTERN = /^#+\s*Acceptance\s+Criteria\s*$/im;
...
+        /\S/u.test(rawCharacter ?? '')
+      ) {
+        if (maskedText[cursor] !== ' ') {
+          fullyMasked = false;
+          break;
+        }
+        sawMaskedCharacter = true;
+      }
+    }
+    if (!fullyMasked || !sawMaskedCharacter) {
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/scripts/suitability-triage.mts]: **Accepted** — fixed in `623794f7`: raw fallback matches are required to fit one code range, with a bounded prefix check for the greedy code-contained occurrence.

File: src/scripts/markdown-code.mts [hunk 8]
```diff
@@ -81,3 +81,158 @@ export function stripMarkdownCodeRegions(text: string): string {
...
+    const newlineIndex = text.indexOf('\n', lineStart);
+    const lineEnd =
+      newlineIndex === -1
+        ? text.length
+        : newlineIndex > lineStart && text[newlineIndex - 1] === '\r'
+          ? newlineIndex - 1
+          : newlineIndex;
+    const lineAfter = newlineIndex === -1 ? text.length : newlineIndex + 1;
+    const line = text.slice(lineStart, lineEnd);
+    const match = line.match(/^ {0,3}(`{3,}|~{3,})(.*)$/u);
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/scripts/markdown-code.mts]: **Accepted** — fixed in `3467f699` by parsing fenced blocks inside blockquote containers, including tilde fences, with focused regression coverage.

File: src/scripts/markdown-code.mts [hunk 9]
```diff
@@ -81,3 +81,191 @@ export function stripMarkdownCodeRegions(text: string): string {
...
+        ranges.push({
+          start: cursor,
+          end: candidate + closingLength,
+        });
+        cursor = candidate + closingLength;
+        closed = true;
+        break;
+      }
+      candidate += closingLength;
+    }
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/scripts/markdown-code.mts]: **Rejected** — this repository intentionally treats a single line break inside an inline code span as a literal space, matching the CommonMark guidance in `.github/copilot-instructions.md` and existing multiline code-span tests; blank lines and Markdown block boundaries already terminate the span.

File: src/scripts/markdown-code.mts [hunk 10]
```diff
@@ -81,3 +81,225 @@ export function stripMarkdownCodeRegions(text: string): string {
...
+  return backslashCount % 2 === 1;
+}
+
+function hasBlankLine(text: string, start: number, end: number): boolean {
+  return /\r?\n[ \t]*\r?\n/u.test(text.slice(start, end));
+}
+
+const MARKDOWN_BLOCK_BOUNDARY_PATTERN =
+  /^[ \t]{0,3}(?:#{1,6}(?:[ \t]|$)|>[ \t]+|(?:[-+*])[ \t]+|\d{1,9}[.)][ \t]+)/mu;
+
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/scripts/markdown-code.mts]: **Accepted** — fixed in `3e1c63f4` by treating compact `>` markers as Markdown block boundaries, with regression coverage.

File: src/scripts/markdown-code.mts [hunk 11]
```diff
@@ -81,3 +81,225 @@ export function stripMarkdownCodeRegions(text: string): string {
...
+  }
+  return backslashCount % 2 === 1;
+}
+
+function hasBlankLine(text: string, start: number, end: number): boolean {
+  return /\r?\n[ \t]*\r?\n/u.test(text.slice(start, end));
+}
+
+const MARKDOWN_BLOCK_BOUNDARY_PATTERN =
+  /^[ \t]{0,3}(?:#{1,6}(?:[ \t]|$)|>[ \t]+|(?:[-+*])[ \t]+|\d{1,9}[.)][ \t]+)/mu;
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/scripts/markdown-code.mts]: **Accepted** — thematic-break crossings are now rejected by `MARKDOWN_BLOCK_BOUNDARY_PATTERN` in `623794f7`. Raw HTML-block parsing is outside #1856’s narrow policy-pattern code-region scope and should be tracked separately if a verified need remains.

File: src/scripts/markdown-code.mts [hunk 12]
```diff
@@ -81,3 +81,225 @@ export function stripMarkdownCodeRegions(text: string): string {
...
+            char: fenceChar,
+            length: marker.length,
+            start: lineStart,
+            containerDepth: match.containerDepth,
+          };
+        }
+      } else if (
+        fenceChar === fence.char &&
+        marker.length >= fence.length &&
+        match.containerDepth === fence.containerDepth &&
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/scripts/markdown-code.mts:350]: **Accepted** — fixed in `623794f7` by terminating a quoted fenced range when its required container prefix disappears.

File: src/scripts/suitability-triage.mts [hunk 13]
```diff
@@ -288,6 +292,55 @@ const ACCEPTANCE_CRITERIA_PATTERN = /^#+\s*Acceptance\s+Criteria\s*$/im;
...
+        /\S/u.test(rawCharacter ?? '')
+      ) {
+        if (maskedText[cursor] !== ' ') {
+          fullyMasked = false;
+          break;
+        }
+        sawMaskedCharacter = true;
+      }
+    }
+    if (!fullyMasked || !sawMaskedCharacter || !isCodeOnlyRange(index, end)) {
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/scripts/suitability-triage.mts]: **Accepted** — fixed in `623794f7` by bounding the raw fallback to the code-contained policy occurrence instead of accepting a greedy match extended into nearby prose.

File: src/scripts/markdown-code.mts [hunk 14]
```diff
@@ -81,3 +81,225 @@ export function stripMarkdownCodeRegions(text: string): string {
...
+
+    while (candidate < end) {
+      if (text[candidate] !== '`') {
+        candidate += 1;
+        continue;
+      }
+      const closingLength = countBackticks(text, candidate, end);
+      if (
+        closingLength === openingLength &&
+        !isEscapedBacktick(text, candidate) &&
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/scripts/markdown-code.mts]: **Accepted** — fixed in `623794f7`; closing delimiters now follow CommonMark’s literal-backslash behavior inside code spans, with regression coverage.

File: src/scripts/markdown-code.mts [hunk 15]
```diff
@@ -81,3 +81,225 @@ export function stripMarkdownCodeRegions(text: string): string {
...
+function findMarkdownCodeRanges(text: string): MarkdownCodeRange[] {
+  const fencedRanges = findFencedCodeRanges(text);
+  const ranges = [...fencedRanges];
+  let cursor = 0;
+
+  for (const fencedRange of fencedRanges) {
+    ranges.push(...findInlineCodeRanges(text, cursor, fencedRange.start));
+    cursor = fencedRange.end;
+  }
+  ranges.push(...findInlineCodeRanges(text, cursor, text.length));
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/scripts/markdown-code.mts:504]: **Rejected** — raw HTML-block parsing is outside #1856’s narrow policy-pattern code-region scope; the repository has no HTML-block parser in this helper. Track it as a separate verified follow-up if product scope requires GFM/HTML semantics.

File: src/scripts/markdown-code.mts [hunk 16]
```diff
@@ -81,3 +81,225 @@ export function stripMarkdownCodeRegions(text: string): string {
...
+  marker: string;
+  info: string;
+  containerDepth: number;
+};
+
+function parseFencedLine(line: string): FencedLine | null {
+  const quoteMatch = line.match(/^ {0,3}(?:(?:> ?)+)(.*)$/u);
+  const containerDepth = quoteMatch
+    ? (quoteMatch[0].match(/>/gu)?.length ?? 0)
+    : 0;
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/scripts/markdown-code.mts]: **Rejected** — counting `>` only in the container prefix is already covered by `623794f7`; the remaining HTML-like info-string case is not a new defect in the current parser. Raw HTML/GFM semantics outside fenced-code container parsing are outside #1856’s narrow scope.

File: src/scripts/markdown-code.mts [hunk 17]
```diff
@@ -81,3 +81,225 @@ export function stripMarkdownCodeRegions(text: string): string {
...
+        candidate += 1;
+        continue;
+      }
+      const closingLength = countBackticks(text, candidate, end);
+      if (
+        closingLength === openingLength &&
+        !isEscapedBacktick(text, candidate) &&
+        !hasMarkdownBlockBoundary(text, contentStart, candidate) &&
+        !hasBlankLine(text, contentStart, candidate)
+      ) {
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/scripts/markdown-code.mts:471]: **Rejected** — GFM table-cell parsing is outside #1856’s narrow CommonMark code-region helper scope. It requires a separately scoped table parser and verified product requirement; it is not silently folded into this policy-pattern change.

**Comment by kurone-kito** [comment]
**Rejected** — coderabbitai[bot] did not review HEAD 623794f71e436ab7ef1c9766fabbf65e3971f4ec (review limit reached / rate limited); this is not a completed review (source: #issuecomment-5179820766)

File: src/scripts/suitability-triage.mts [hunk 18]
```diff
@@ -288,6 +292,70 @@ const ACCEPTANCE_CRITERIA_PATTERN = /^#+\s*Acceptance\s+Criteria\s*$/im;
...
+      continue;
+    }
+    const end = index + match[0].length;
+    const codeRange = getCodeRangeAt(index);
+    if (codeRange) {
+      const codeOnlyMatch = POLICY_OVERRIDE_PATTERN.exec(
+        text.slice(index, codeRange.end),
+      );
+      if (codeOnlyMatch?.index === 0) {
+        continue;
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/scripts/suitability-triage.mts:338]: **Accepted** — fixed in `3466edad` by resuming the raw regex at the end of only the code-contained range; the new regression keeps a later partially formatted prose directive failing Check 3.

File: src/scripts/markdown-code.mts [hunk 19]
```diff
@@ -81,3 +81,238 @@ export function stripMarkdownCodeRegions(text: string): string {
...
+  }
+  return backslashCount % 2 === 1;
+}
+
+function hasBlankLine(text: string, start: number, end: number): boolean {
+  return /\r?\n[ \t]*\r?\n/u.test(text.slice(start, end));
+}
+
+const MARKDOWN_BLOCK_BOUNDARY_PATTERN =
+  /^[ \t]{0,3}(?:#{1,6}(?:[ \t]|$)|>[ \t]*|(?:[-+*])[ \t]+|\d{1,9}[.)][ \t]+|(?:-{3,}|_{3,}|\*{3,})[ \t]*$)/mu;
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/scripts/markdown-code.mts]: **Accepted** — fixed in `3466edad` by recognizing setext `=` and `-` underlines as block boundaries, with regression coverage alongside the existing thematic-break cases.

File: src/scripts/markdown-code.mts [hunk 20]
```diff
@@ -81,3 +81,238 @@ export function stripMarkdownCodeRegions(text: string): string {
...
+      cursor = contentStart;
+    }
+  }
+
+  return ranges;
+}
+
+function findMarkdownCodeRanges(text: string): MarkdownCodeRange[] {
+  const fencedRanges = findFencedCodeRanges(text);
+  const ranges = [...fencedRanges];
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/scripts/markdown-code.mts]: **Accepted** — fixed in `3466edad` by collecting four-space/tab-indented Markdown code ranges and adding an indented-code false-positive regression.

File: src/scripts/markdown-code.mts (hunk 19 above)
[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/scripts/markdown-code.mts]: **Accepted** — fixed in `3466edad` by tracking the continuing blockquote container depth when checking inline-span boundaries, with a multiline quoted-span regression.

**Comment by kurone-kito** [comment]
_codex-848483b6-cabc-48e4-8d2c-77182322ec9d: review triage snapshot — IDD automation marker. Do not edit._

File: src/scripts/markdown-code.mts [hunk 21]
```diff
@@ -81,3 +81,340 @@ export function stripMarkdownCodeRegions(text: string): string {
...
+function findIndentedCodeRanges(text: string): MarkdownCodeRange[] {
+  const ranges: MarkdownCodeRange[] = [];
+  let rangeStart: number | null = null;
+  let rangeEnd = 0;
+  let lineStart = 0;
+
+  while (lineStart <= text.length) {
+    const line = lineBounds(text, lineStart);
+    const parsed = parseContainerLine(text.slice(lineStart, line.end));
+    const isIndented = /^(?: {4,}|\t)/u.test(parsed.content);
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/scripts/markdown-code.mts]: **Accepted** — fixed in `63473a5e`/`f3dfdfc3` by requiring a blank line, prior block boundary, active indented range, or container transition before starting an indented code range; the paragraph-continuation regression keeps the visible directive failing.

File: src/scripts/suitability-triage.mts [hunk 22]
```diff
@@ -453,12 +530,34 @@ export function checkTrustSafety(context: Context): CheckOutcome {
...
+  // even when a fenced block precedes the match.
+  const bodyOffset = issue.title.length + 1;
+  const policyMatch = findPolicyOverrideMatch(
+    corpus,
+    `${issue.title}\n${maskMarkdownCodeRegionsPreservingPositions(issue.body)}`,
+    (start) => {
+      if (start < bodyOffset) {
+        return null;
+      }
+      const range = getMarkdownCodeRange(issue.body, start - bodyOffset);
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/scripts/suitability-triage.mts]: **Accepted** — fixed in `f3dfdfc3` by computing body code ranges once, reusing them for masking and raw matching, and using binary search for range lookup; the 6,000-span check measured about 69ms locally.

**Comment by kurone-kito** [comment]
_codex-848483b6-cabc-48e4-8d2c-77182322ec9d: review triage snapshot — IDD automation marker. Do not edit._

File: src/scripts/markdown-code.mts [hunk 23]
```diff
@@ -81,3 +81,369 @@ export function stripMarkdownCodeRegions(text: string): string {
...
+      rangeEnd = line.next;
+    } else if (rangeStart !== null) {
+      ranges.push({ start: rangeStart, end: rangeEnd });
+      rangeStart = null;
+      rangeEnd = 0;
+    }
+
+    previousLineBlank = isBlank;
+    previousLineBlockBoundary =
+      MARKDOWN_BLOCK_CONTENT_PATTERN.test(parsed.content) ||
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/scripts/markdown-code.mts]: **Accepted** — fixed in `14164900` by separating indented-code start preceders from list-item paragraph boundaries; a list-item continuation regression keeps visible directives failing.

File: src/scripts/markdown-code.mts [hunk 24]
```diff
@@ -81,3 +81,369 @@ export function stripMarkdownCodeRegions(text: string): string {
...
+
+    while (candidate < end) {
+      if (text[candidate] !== '`') {
+        candidate += 1;
+        continue;
+      }
+      const closingLength = countBackticks(text, candidate, end);
+      if (
+        closingLength === openingLength &&
+        !hasMarkdownBlockBoundary(text, contentStart, candidate) &&
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/scripts/markdown-code.mts]: **Accepted** — fixed in `14164900` by stopping unmatched inline-span searches at the next block boundary; the 8,000-line regression measured about 16ms locally.

File: src/scripts/markdown-code.mts [hunk 25]
```diff
@@ -81,3 +81,369 @@ export function stripMarkdownCodeRegions(text: string): string {
...
+  }
+  return backslashCount % 2 === 1;
+}
+
+function hasBlankLine(text: string, start: number, end: number): boolean {
+  return /\r?\n[ \t]*\r?\n/u.test(text.slice(start, end));
+}
+
+const MARKDOWN_BLOCK_CONTENT_PATTERN =
+  /^(?:#{1,6}(?:[ \t]|$)|(?:[-+*])[ \t]+|\d{1,9}[.)][ \t]+|(?:-{1,}|={1,}|_{3,}|\*{3,})[ \t]*$)/u;
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/scripts/markdown-code.mts]: **Accepted** — fixed in `14164900` by limiting paragraph-interrupting ordered markers to start number 1, with a non-one-marker multiline-span regression.

**Comment by kurone-kito** [comment]
_codex-848483b6-cabc-48e4-8d2c-77182322ec9d: review triage snapshot — IDD automation marker. Do not edit._

File: src/scripts/suitability-triage.mts [hunk 26]
```diff
@@ -288,6 +293,79 @@ const ACCEPTANCE_CRITERIA_PATTERN = /^#+\s*Acceptance\s+Criteria\s*$/im;
...
+    const codeRange = getCodeRangeAt(index);
+    if (codeRange) {
+      const codeOnlyMatch = POLICY_OVERRIDE_PATTERN.exec(
+        text.slice(index, codeRange.end),
+      );
+      if (codeOnlyMatch?.index === 0) {
+        // The raw pattern may greedily span a code-only occurrence and a
+        // later prose occurrence. Resume just after the inert range so the
+        // later occurrence is evaluated independently.
+        pattern.lastIndex = codeRange.end;
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/scripts/suitability-triage.mts]: **Accepted** — fixed in `3e2f3609` by resuming raw policy matching after only the inert code-contained occurrence, so a later trigger in the same code range can still match visible prose.

File: src/scripts/markdown-code.mts [hunk 27]
```diff
@@ -81,3 +81,372 @@ export function stripMarkdownCodeRegions(text: string): string {
...
+  const quoteMatch = line.match(/^ {0,3}(?:(?:> ?)+)(.*)$/u);
+  return {
+    content: quoteMatch ? quoteMatch[1] : line,
+    containerDepth: quoteMatch ? (quoteMatch[0].match(/>/gu)?.length ?? 0) : 0,
+  };
+}
+
+function parseFencedLine(line: string): FencedLine | null {
+  const { content, containerDepth } = parseContainerLine(line);
+  const fenceMatch = content.match(/^ {0,3}(`{3,}|~{3,})(.*)$/u);
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/scripts/markdown-code.mts]: **Accepted** — fixed in `3e2f3609` by stripping list-item container prefixes before fence parsing and adding list-fence regressions for both masking and Check 3.

**Comment by kurone-kito** [comment]
_codex-848483b6-cabc-48e4-8d2c-77182322ec9d: review triage snapshot — IDD automation marker. Do not edit._

File: src/scripts/markdown-code.mts [hunk 28]
```diff
@@ -81,3 +84,384 @@ export function stripMarkdownCodeRegions(text: string): string {
...
+          : newlineIndex;
+    const lineAfter = newlineIndex === -1 ? text.length : newlineIndex + 1;
+    const line = text.slice(lineStart, lineEnd);
+    const containerLine = parseContainerLine(line);
+    const match = parseFencedLine(line);
+
+    if (
+      fence !== null &&
+      fence.containerDepth > 0 &&
+      containerLine.containerDepth < fence.containerDepth
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/scripts/markdown-code.mts]: **Accepted** — fixed in `2099eb2f` by tracking the list item content indentation and terminating a list fence before an unindented top-level line.

File: src/scripts/markdown-code.mts [hunk 29]
```diff
@@ -25,30 +25,33 @@
...
-    // CommonMark §4.5: a fence marker may be indented by at most three spaces;
-    // a marker with four or more leading spaces is an indented code line, not a
-    // fence, so accepting arbitrary leading whitespace here would wrongly enter
-    // fence mode on `    ~~~` and blank the real content that follows it.
-    const openMatch = line.match(/^ {0,3}(`{3,}|~{3,})(.*)$/);
-    if (openMatch) {
-      const marker = openMatch[1];
-      const info = openMatch[2];
-      const fenceChar = marker[0];
+    const parsed = parseFencedLine(line);
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/scripts/markdown-code.mts:36]: **Accepted** — fixed in `2099eb2f` by tracking the required blockquote depth in `blankFencedCodeBlocks` and ending the mask when that quote container disappears.

File: src/scripts/markdown-code.mts [hunk 30]
```diff
@@ -81,3 +84,384 @@ export function stripMarkdownCodeRegions(text: string): string {
...
+    } else if (rangeStart !== null) {
+      ranges.push({ start: rangeStart, end: rangeEnd });
+      rangeStart = null;
+      rangeEnd = 0;
+    }
+
+    previousLineBlank = isBlank;
+    previousLineBlockBoundary =
+      MARKDOWN_INDENTED_CODE_PRECEDER_PATTERN.test(parsed.content) ||
+      parseFencedLine(rawLine) !== null;
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/scripts/markdown-code.mts]: **Accepted** — fixed in `2099eb2f` by requiring a valid fence opener before an indented-code boundary can begin; backtick info strings are now rejected consistently.

File: src/scripts/markdown-code.mts [hunk 31]
```diff
@@ -81,3 +84,384 @@ export function stripMarkdownCodeRegions(text: string): string {
...
+  let previousLineBlank = true;
+  let previousLineBlockBoundary = true;
+  let previousContainerDepth = 0;
+  let lineStart = 0;
+
+  while (lineStart <= text.length) {
+    const line = lineBounds(text, lineStart);
+    const rawLine = text.slice(lineStart, line.end);
+    const parsed = parseContainerLine(rawLine);
+    const isIndented = /^(?: {4,}|\t)/u.test(parsed.content);
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/scripts/markdown-code.mts]: **Accepted** — fixed in `2099eb2f` by expanding leading tabs to four-column stops when classifying indented code, covering space-prefixed tab indentation.

**Comment by kurone-kito** [comment]
_codex-848483b6-cabc-48e4-8d2c-77182322ec9d: review triage snapshot — IDD automation marker. Do not edit._
